### PR TITLE
Add lab1 - Fibonacci series

### LINKED ISSUE #591: Lab1 - Fibonacci series
- Implement Fibonacci series
- Implement Negative Fibonacci series
- Add unit tests

### PR DESCRIPTION
**Pull request by harirakr**
Fixes #591

Review of colleague's PR #587 

#### Changes proposed in this PR:
- Implement Fibonacci and Negative Fibonacci series
- Add unit tests

This was reviewed before - PR #588 . But while incorporating the comments, I made a few incorrect merge/rebase operations and now the changes are not good to be merged. I will close PR #588 , if it's alright.

### CODE REVIEW

File: 01_fib/harirakr/main_test.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	main()
+	expected := strconv.Quote("[1 1 2 3 5 8 13]\n")
+	actual := strconv.Quote(buf.String())
+	if expected != actual {
+		t.Errorf("main fail; expected=%v, actual=%v", expected, actual)
+	}
+}
+
+// Test input out of bounds
+func TestFibOutOfBounds(t *testing.T) {
```
Comment (n0npax) [inline 01_fib/harirakr/main_test.go:37]: The code here is duplicated.
I'd prefer to have a loop inside this test.

sth like:
```
for _, val := range []int{-94,-93,93,93} {
        var buf bytes.Buffer
	out = &buf
	fib(val)
	expected := strconv.Quote("fib: Range out of bounds; takes input > -93 and < 93\n")
	actual := strconv.Quote(buf1.String())
	if expected != actual {
		t.Errorf("Test fail fib(%d). Expected=%v, Actual=%v", val, expected, actual)
	}
}
```
or even more fancy:
```
for _, val := range []int{-94,-93,93,93} {
  t.Run(fmt.Sprintf("val%s", val), func(t *testing.T) {
        var buf bytes.Buffer
	out = &buf
	fib(val)
	expected := strconv.Quote("fib: Range out of bounds; takes input > -93 and < 93\n")
	actual := strconv.Quote(buf1.String())
	if expected != actual {
		t.Errorf("Test fail fib(%d). Expected=%v, Actual=%v", val, expected, actual)
	}
  })
}
```

File: 01_fib/harirakr/main_test.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	assert.Equal(t, []int64{}, fib(0))
+}
+
+func TestMain(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+	main()
+	expected := strconv.Quote("[1 1 2 3 5 8 13]\n")
+	actual := strconv.Quote(buf.String())
+	if expected != actual {
```
Comment (n0npax) [inline 01_fib/harirakr/main_test.go]: you're using `assert.Equal` above. Why not use it also here? I believe it's a better fit.

**Review by n0npax — CHANGES_REQUESTED** [review]
good work mate. Just 2 minor things in tests.

File: 01_fib/harirakr/main_test.go (hunk 1 above)
Comment (n0npax) [reply 01_fib/harirakr/main_test.go:37]: and  `	if expected != actual {` -> `assert.Equal`

Comment (harirakr) [reply 01_fib/harirakr/main_test.go:37]: Don. 
Thanks @n0npax

File: 01_fib/harirakr/main_test.go (hunk 2 above)
Comment (harirakr) [reply 01_fib/harirakr/main_test.go]: Don.

**Review by n0npax — APPROVED** [review]
lgtm

File: 01_fib/harirakr/main.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+		fmt.Fprintln(out, "fib: Range out of bounds; takes input > -93 and < 93")
+		return nil
+	}
+
+	var sign int64 = 1
+	if n < 0 {
+		sign = -1
+	}
+
+	var a, b, i int64
```
Comment (anzboi) [inline 01_fib/harirakr/main.go:22]: Wouldn't bother declaring any of these, the following works just fine
```go
a, b := 0, 1

...

for i := 0; i < n * sign; i++ {
        ...
}
```

File: 01_fib/harirakr/main_test.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import (
+	"bytes"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNegativeFib(t *testing.T) {
+	assert.Equal(t, []int64{1, -1, 2, -3, 5, -8, 13, -21, 34, -55}, fib(-10))
```
Comment (anzboi) [inline 01_fib/harirakr/main_test.go:12]: Great test coverage

**Review by anzboi — APPROVED** [review]
LGTM
Left a comment about declaring variables unecessarily, usually in Go you avoid unecessary declarations unless you have to but its not a big deal.

File: 01_fib/harirakr/main.go (hunk 3 above)
Comment (harirakr) [reply 01_fib/harirakr/main.go:22]: Sure, will keep this in mind. Thanks

File: 01_fib/harirakr/main_test.go (hunk 4 above)
Comment (harirakr) [reply 01_fib/harirakr/main_test.go:12]: Thanks to @juliaogris  for PR# 588

**Pull request merged**